waveform_editor/configuration.py

```diff
@@ -216,6 +293,24 @@ def _to_commented_map(self):
             result[group_name] = group.to_commented_map()
         return result
 
+    def _calculate_bounds(self):
+        min_start = float("inf")
+        max_end = float("-inf")
+
+        for name in self.waveform_map:
+            waveform = self[name]
+            if not isinstance(waveform, DerivedWaveform) and waveform.tendencies:
+                start = waveform.tendencies[0].start
+                end = waveform.tendencies[-1].end
+
+                if start < min_start:
+                    min_start = start
+                if end > max_end:
+                    max_end = end
```

Comment:
You can make this a bit shorter
```suggestion
                min_start = min(min_start, waveform.tendencies[0].start)
                max_end = max(max_end, waveform.tendencies[-1].end)
```